### DISCUSSION TITLE
How to tint Container ?

### DISCUSSION THREAD

**Discussion by humbleCoder1990** [pixijs/pixijs#7765] [Q&A]
I have a Container which has 3000 sprits from Texture.WHITE. 

To change the color, iterating all sprites's tint property seems overkill.
Because the children colors will be all same based Texture.WHITE.
I just want to tint like container.tint = 0xff0000.
I want to know what performant way is.
Because I'll change color so often.

**Comment by ivanpopelyshev** [comment]
Iterate through all sprites, there's no other way.
If you add tint to container - you will have to iterate through all sprites anyway :)

> I want to know what performant way is.
Like a for of 3000 elements with assignment.

Compare: to actually draw those sprites in one frame, pixi has to iterate through all of them, and do interlave: https://github.com/pixijs/pixijs/blob/7ee2be5cf88bb990166db10cfb017e37aa2cbbb1/packages/core/src/batch/AbstractBatchRenderer.ts#L746-L781

Compared to that, your FOR is small

**Reply by humbleCoder1990** [reply]
Wow..Thank Ivan always for amazing tutorial....
I'm getting to know how PIXI actually works under the hood :)
What's your opinion about the meaning "A lot Sprites" in PIXI ? (in one frame)
If 3000 is small, 10000 is ok?

**Comment by miltoncandelero** [comment]
Don't know if this would be a good approach but the ColorMatrixFilter has a tint method
